### PR TITLE
Add an image resolver for Debian.

### LINKED ISSUE #740: Add an image resolver for debian
Just like ubuntu.

### PR DESCRIPTION
**Pull request by mikalstill**
Fixes #740.

### CODE REVIEW

**Comment by mikalstill** [comment]
CI still doesn't work, so ignore that failure please.

File: shakenfist/image_resolver_cirros.py [hunk 1]
```diff
@@ -31,36 +33,16 @@ def resolve(name):
...
-            break
-    if not checksum_url:
-        log.warning('Did not find checksum')
-
-    log.with_field('checksum', checksum).info('Checksum retrieval')
-
+    checksums = util.fetch_remote_checksum(checksum_url)
+    checksum = checksums.get(os.path.basename(url))
+    LOG.with_field('name', name).with_field('url', url).with_field(
+        'checksum', checksum).info('Image resolved')
```
Comment (mandoonandy) [inline shakenfist/image_resolver_cirros.py:47]: `with_fields()` would reduce verbosity here.

File: shakenfist/image_resolver_debian.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        checksum_url = CHECKSUM_URL % {'version': vernum}
+        resp = requests.head(url,
+                             headers={'User-Agent': util.get_user_agent()})
+        if resp.status_code != 200:
+            raise exceptions.HTTPError(
+                'Failed to fetch %s, status code %d' % (url, resp.status_code))
+
+    else:
+        found_any = False
+        for vernum in range(9, 20):
```
Comment (mandoonandy) [inline shakenfist/image_resolver_debian.py:36]: Would there be less lines of code if you searched backwards and stopped when a valid URL was found?

**Review by mandoonandy — CHANGES_REQUESTED** [review]
One question (probably misunderstanding the code) and one suggestion.

File: shakenfist/image_resolver_debian.py (hunk 2 above)
Comment (mikalstill) [reply shakenfist/image_resolver_debian.py:36]: So, Debian only has 9 and 10 online at the moment. 10 is the current stable. So this is more future proofing than anything else. I guess I could load the index page and parse, but I was being lazy.

File: shakenfist/image_resolver_cirros.py (hunk 1 above)
Comment (mikalstill) [reply shakenfist/image_resolver_cirros.py]: Done.

**Review by mandoonandy — APPROVED** [review]
LGTM

**Pull request merged**